[WASI] Compilation failure on Swift 6.1+ due to missing pthread symbols in WASI toolchains

### Summary
In `swift-system` 1.8.0, compiling for WebAssembly target (`wasm32-unknown-wasi`) on Swift 6.1+ fails with scope resolution errors for POSIX pthread APIs (`pthread_setspecific`, `pthread_getspecific`, `pthread_key_create`).

### Details
In `Sources/System/Internals/Exports.swift`, TLS storage for WASI is gated on:
```swift
#elseif os(WASI) && (swift(<6.1) || !_runtime(_multithreaded))
```

When compiling with Swift 6.1+ on a standard WASI SDK (`wasm32-unknown-wasi`):
1. `swift(<6.1)` evaluates to `false`.
2. The compiler falls into the `#else` branch which assumes POSIX `pthread_key_t`, `pthread_key_create`, `pthread_setspecific`, and `pthread_getspecific` exist.
3. Standard WASI libc does not expose POSIX pthread functions in scope for non-threaded WASI target toolchains.

### Compilation Errors
```
Sources/System/Internals/Exports.swift:221:14: error: cannot find 'pthread_key_create' in scope
Sources/System/Internals/Exports.swift:233:14: error: cannot find 'pthread_setspecific' in scope
Sources/System/Internals/Exports.swift:242:10: error: cannot find 'pthread_getspecific' in scope
```

### Expected Behavior
`swift-system` should continue using the single-threaded WASI TLS fallback when POSIX pthread APIs are not available in scope for WASI targets.

--- a/Sources/System/Internals/Exports.swift
+++ b/Sources/System/Internals/Exports.swift
@@ -184,7 +184,7 @@ extension String {
 // TLS
 #if os(Windows)
 internal typealias _PlatformTLSKey = DWORD
-#elseif os(WASI) && (swift(<6.1) || !_runtime(_multithreaded))
+#elseif os(WASI) && !_runtime(_multithreaded)
 // Mock TLS storage for single-threaded WASI.
 // Carries no state of its own, and only used to index `sharedTLSStorage`.
 internal final class _PlatformTLSKey: @unchecked Sendable {
@@ -214,7 +214,7 @@ internal func makeTLSKey() -> _PlatformTLSKey {
     fatalError("Unable to create key")
   }
   return raw
-  #elseif os(WASI) && (swift(<6.1) || !_runtime(_multithreaded))
+  #elseif os(WASI) && !_runtime(_multithreaded)
   return _PlatformTLSKey()
   #else
   var raw = pthread_key_t()

